[문승준] Mission 6. 로그인 JWT 적용

palza4dev opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
- 로그인 성공시 JWT로 access_token을 생성하여 전달
		 
## 수정 사항들 자세한 내용
- JWT 설치 후 import
- 암호화된 비밀번호 일치 비교를 위해 bcrypt.checkpw() 함수 사용
- 토큰 생성을 위해 SECRET_KEY import, 알고리즘은 HS256을 사용함
- 로그인 성공시 response body에 토큰 전달함

## 기타 질문 및 특이 사항
- 토큰을 http 응답/요청의 headers 부분에 넣는 것으로 알고 있는데 이에 대해 더 알아보겠습니다.
- Client에게 토큰을 전달 받아서 서버에서 인증 및 인가를 하는 과정에 대해 더 알아보겠습니다. 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

## POSTMAN을 이용한 로그인 성공시 token 전달 테스트
![스크린샷 2021-09-29 오후 6 52 52](https://user-images.githubusercontent.com/72376931/135247971-ae17b1c7-60e7-40a4-a5bb-84bcd59942a1.png)

palza4dev committed:
<Docs> requirements.txt에 PyJWT 버전 추가 기입

palza4dev committed:
<Add> 로그인시 비밀번호 검증과 jwt 토큰 생성
- bcrypt.checkpw()로 로그인시 암호화된 비밀번호 일치여부 확인
- jwt 패키지 설치하여 로그인 성공시 access_token 전달

kyeongraekim commented on students/seungjun/users/views.py:
```diff
@@ -56,11 +57,12 @@ def post(self,request):
+            token = jwt.encode({'id': user.id}, SECRET_KEY, algorithm = 'HS256')
+            return JsonResponse({'access_token': token,'message':'SUCCESS'}, status=200)
             
         except KeyError:
```
키에러를 제외한 (오늘 세션에서 언급된) 발생 가능한 다양한 예외상황들도 한 번 핸들링 해보시는 것을 권장드립니다~
한 가지 예를 들자면, 해당되는 객체가 가 없을 때 발생할 수 있는 DoesNotExist 에러가 있습니다.

palza4dev replied:
DoesNotExist 에러와 JSONDecodeError 핸들링을 위한 except 구문을 추가하였습니다.
따라서, try문 안에 있던 INVALID_EMAIL 에러핸들링은 삭제하였습니다.
except문에 예외처리를 명시하는 것이 더 나은 방법이라 생각했습니다.

kyeongraekim replied:
LGTM!~

kyeongraekim reviewed (changes requested):
승준님 오늘 세션에 맞게 코드 잘 작성해 주신 것 확인했습니다~
아래 한 가지 추가 반영사항 확인해주셔서 수정 도전해보세요!~

palza4dev committed:
<Modify> 에러핸들링 추가
- SingUpView에 JSONDecodeError 예외처리 추가
- LogInView에 JSONDcodeError 예외처리 추가
- try문이 아닌 except문에서 DoesNotExist 예외처리로 email 검증
- 명시적이고 구체적인 예외처리를 위함

palza4dev committed:
<Fix> SECRET_KEY import 경로수정

kyeongraekim commented:
LGTM!~ 이번 위스타그램 내용 블로그에 정리해보면서 꼭 다시 한 번 꼼꼼이 복습 해보세요!~
